### PR TITLE
upgraded library json-path-upgrade

### PR DESCRIPTION
**Pull request by Cjpilbdev**
upgrade library because of vulnerabilities.

### CODE REVIEW

File: deploy-service/common/pom.xml [hunk 1]
```diff
@@ -112,7 +112,12 @@
         <dependency>
             <groupId>com.jayway.jsonpath</groupId>
             <artifactId>json-path</artifactId>
-            <version>0.9.1</version>
+            <version>2.9.0</version>
+        </dependency>
+        <dependency>
+            <groupId>net.minidev</groupId>
+            <artifactId>json-smart</artifactId>
+            <version>2.5.1</version>
```
Comment (gzpcho) [inline deploy-service/common/pom.xml:120]: Any breaking changes that might impact existing functionality?

**Comment by tylerwowen** [comment]
I was thinking of switching to jackson for json processing but that requires code change. If we need code change anyways, then we should just reimplement with jackson and remove the dependency.

File: deploy-service/common/src/main/java/com/pinterest/deployservice/common/DBConfigReader.java [hunk 2]
```diff
@@ -81,10 +83,22 @@ public Integer getPort(String replicaSetName) {
     }
 
     public String getUsername(String role) {
-        return JsonPath.read(credJson, String.format("$.%s.users[?(@.enabled == true)][0].username", role));
+        JSONArray jsonArray = JsonPath.read(credJson, String.format("$.%s.users[?(@.enabled == true)]", role));
+        if(jsonArray.size()>0){
+            Map hashMap =  (Map) jsonArray.get(0);
```
Comment (gzpcho) [inline deploy-service/common/src/main/java/com/pinterest/deployservice/common/DBConfigReader.java:88]: Should we not initialize a new `HashMap` with the `new` keyword? What's the structure of `jsonArray.get(0)`?

File: deploy-service/common/pom.xml (hunk 1 above)
Comment (Cjpilbdev) [reply deploy-service/common/pom.xml:120]: this is only used for getting the database configuration from json  in DBConfigReader

**Review by tylerwowen — COMMENTED** [review]
LGTM.
Can you run `mvn spotless:apply` and only commit the changes to `deploy-service/common/src/main/java/com/pinterest/deployservice/common/DBConfigReader.java`?

**Pull request merged**